Controllers/StaffController.cs

```diff
@@ -27,10 +27,10 @@ public async Task<IActionResult> Index(string searchString)
 
             if (!String.IsNullOrEmpty(searchString))
             {
-                staff = staff.Where(s => s.LastName.Contains(searchString));
+                staff = staff.Where(s => s.FullName.Contains(searchString));
             }
-
-            return View(await staff.ToListAsync());
+            
+            return View(await _context.Staff.ToListAsync());
```

Comment:
Thinking of single responsibility principle, a class/function should have one job, and do that one job well. In this case you have the controller acting as both a manager, and a respository for queries/data access. I would suggest you create a repository class for staff, and all of the queries related to staff. Giving you a good separation of control.